### ISSUE TITLE
No API key version negotiation

### ISSUE THREAD

**Issue by ppatierno** [IBM/sarama#1916]
If I got it correctly, the Sarama client doesn't have any kind of API version negotiation in order to handle backward compatibility automatically (as it happens in Java clients).
It seems to rely on the `Version` specified in the configuration.
Even if the `Broker` struct provides an `ApiVersions` function, it seems to be not used by the client to get the min/max version for each API key and then handling the communication with the broker based on those versions.
Is there any specific reason for that?

**Comment by ppatierno** [comment]
@dnwe thanks for the update!

**Comment by dnwe** [comment]
As mentioned, this is planned for a future v2 version. The plumbing is in-place and I have experimented with some drafts in the past. Marking the issue as stale/exempt

**Comment by ijuma** [comment]
Isn't this completed now, can we close it?

**Comment by dnwe** [comment]
Yes this was completed in https://github.com/IBM/sarama/releases/tag/v1.46.0
